### DISCUSSION TITLE
why my foreign key values always comes null?  somebody will be able to help me with this please?

### DISCUSSION THREAD

**Discussion by justnashy** [community/community#133346] [Programming Help]
### Body

HELLO ALL,

I AM STUCK WITH UNIVERSITY ASSESSMENT AND NEW TO MYSQL.

WHEN I TRY TO DISPLAY TABLE DATA MY FOREIGN KEY VALUES ALWAYS COMES AS NULL,
I HAVE NOT BEEN DELETED ANY OF PARENT TABLE AND HERE ARE THE CODES,
COULD SOMEONE ASSIST ME WITH FIX THIS UP FOR ME PLEASE?

CHILD TABLE:

CREATE TABLE PLAYERS (
FIRST_NAME VARCHAR(30) NOT NULL,
LAST_NAME VARCHAR(30) NOT NULL,
GENDER ENUM('M','F') NOT NULL,
PLAYER_ID INT, FOREIGN KEY (PLAYER_ID) REFERENCES PLAYERS_NEW(PLAYER_ID)
);

PARENT TABLE:

CREATE TABLE PLAYERS_NEW (
PLAYER_ID INT PRIMARY KEY AUTO_INCREMENT NOT NULL,
FIRST_NAME_X VARCHAR(30) NOT NULL,
LAST_NAME_X VARCHAR(30) NOT NULL,
GENDER_X ENUM ('M','F') NOT NULL
);

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by joviprata** [comment]
Hey justnashy,
A foreign key is usually set to null if the row in the child table doesn’t find a row in the parent table that matches one or several specific column values. For this case, I’m assuming the child only needs to match the column regarding the last name of the parent (LAST_NAME equal to LAST_NAME_X) in order to be considered a child of that parent.
Also, it’s important that you create the parent table first, then the child table, so that the foreign key can use the primary key as reference. And the parents table is usually populated first with rows, so that they can be used as reference for the child table when assigning the foreign key values for its rows.
Hope this helps.
